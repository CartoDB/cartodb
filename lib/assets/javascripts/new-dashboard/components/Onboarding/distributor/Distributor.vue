<template>
  <section class="onboarding-welcome">
    <Modal name="onboarding-welcome" :isOpen="true" @closeModal="closeModal">
      <div class="distributor gtm-onboarding">
        <div class="header">
          <div class="container">
            {{ $t('Wizards.Distributor.headerTitle') }}
          </div>
        </div>

        <div class="container u-pt--36">
          <h1 class="u-pr--10 u-pl--10 u-mb--40 title is-body is-semibold">
            {{ $t('Wizards.Distributor.title') }}
          </h1>

          <div class="grid u-flex__justify--between">
            <div class="grid-cell grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile">
              <Selector
                :title="$t('Wizards.cartovl.title')"
                :text="$t('Wizards.cartovl.subtitle')"
                :tags="$t('Wizards.cartovl.tags')"
                @click.native="openOnboarding('carto-vl')"
                iconModifier="map"></Selector>
              <Warning v-if="!hasEngine"></Warning>
            </div>

            <div class="grid-cell grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile">
              <Selector
                :title="$t('Wizards.cartoframes.title')"
                :text="$t('Wizards.cartoframes.subtitle')"
                :tags="$t('Wizards.cartoframes.tags')"
                @click.native="openOnboarding('cartoframes')"
                iconModifier="notebook"></Selector>
              <Warning v-if="!hasEngine"></Warning>
            </div>

            <div class="grid-cell grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile">
              <Selector
                :title="$t('Wizards.builder.title')"
                :text="$t('Wizards.builder.subtitle')"
                :tags="$t('Wizards.builder.tags')"
                @click.native="openOnboarding('builder')"
                iconModifier="window"></Selector>
            </div>
          </div>
          <p class="grid-cell u-mt--48 is-small">{{ $t('Wizards.Distributor.feedback.text') }}
            <a href="https://docs.google.com/forms/d/e/1FAIpQLScBQUWd-TP3Qy514DOCNg-KoLrViHijUR5giLAMS-3jmDnrPg/viewform" target="_blank" rel="noopener noreferrer">
            {{ $t('Wizards.Distributor.feedback.link') }}</a>
          </p>
        </div>
      </div>
    </Modal>

    <router-view name="onboarding" />
  </section>
</template>

<script>
import Selector from 'new-dashboard/components/Onboarding/distributor/Selector';
import Warning from 'new-dashboard/components/Onboarding/distributor/Warning';
import Modal from 'new-dashboard/components/Modal.vue';

export default {
  name: 'OnboardingWelcome',
  components: {
    Selector,
    Warning,
    Modal
  },
  computed: {
    hasEngine () {
      return this.$store.getters['user/hasEngine'];
    }
  },
  methods: {
    openOnboarding (onboardingId) {
      this.$router.push({
        name: 'onboarding-open',
        params: { onboardingId },
        hash: '#step-1'
      });
    },
    closeModal () {
      this.$router.push({ name: 'home' });
    }
  }
};
</script>
<style scoped lang="scss">
@import 'new-dashboard/styles/variables';

.distributor {
  height: 100%;
  background-color: $onboarding__bg-color;
  color: $text__color;
}

.header {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 98px;
  background: $white;
}

.button {
  &.button--ghost {
    margin-right: 36px;
    padding: 0;
  }
}
</style>
