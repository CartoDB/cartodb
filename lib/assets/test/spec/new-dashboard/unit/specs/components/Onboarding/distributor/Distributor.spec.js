import { shallowMount } from '@vue/test-utils';
import Distributor from 'new-dashboard/components/Onboarding/distributor/Distributor';

const $t = key => key;

describe('Distributor.vue', () => {
  it('should render properly', () => {
    const $store = {
      getters: {
        'user/hasEngine': true
      }
    };

    const distributorComponent = createDistributorComponent({
      mocks: {
        $store
      }
    });

    expect(distributorComponent).toMatchSnapshot();
  });

  it('should render engine warning for no engine users', () => {
    const distributorComponent = createDistributorComponent();
    expect(distributorComponent).toMatchSnapshot();
  });

  describe('Methods', () => {
    it('should call window.open with _blank when opening any onboarding', () => {
      global.open = jest.fn();

      const Selector = {
        render: h => h('div'),
        name: 'Selector'
      };

      const distributorComponent = createDistributorComponent({
        stubs: { Selector }
      });

      distributorComponent.find({ name: 'Selector' }).trigger('click');

      expect(global.open).toHaveBeenCalledWith('https://docs.carto.com/spatial-extension-bq/tiler/guides/', '_blank');
    });
  });

  it('should call $router.push when closing modal', () => {
    const $router = {
      push: jest.fn()
    };

    const distributorComponent = createDistributorComponent({
      mocks: { $router }
    });

    distributorComponent.vm.closeModal();

    expect($router.push).toHaveBeenCalledWith({ name: 'home' });
  });
});

function createDistributorComponent (overrides = {}) {
  const $store = {
    getters: {
      'user/hasEngine': false
    }
  };

  const distributorComponent = shallowMount(Distributor, {
    ...overrides,
    mocks: {
      $t,
      $store,
      ...overrides.mocks
    }
  });

  return distributorComponent;
}
