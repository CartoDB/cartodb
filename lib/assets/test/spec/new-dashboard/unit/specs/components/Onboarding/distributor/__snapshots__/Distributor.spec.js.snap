// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Distributor.vue should render engine warning for no engine users 1`] = `
<section class="onboarding-welcome">
  <modal-stub name="onboarding-welcome" isopen="true">
    <div class="distributor gtm-onboarding">
      <div class="header">
        <div class="container">
          Wizards.Distributor.headerTitle
        </div>
      </div>
      <div class="container u-pt--36">
        <h1 class="u-pr--10 u-pl--10 u-mb--40 title is-body is-semibold">
          Wizards.Distributor.title
        </h1>
        <div class="grid u-flex__justify--between">
          <div class="grid-cell grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile">
            <selector-stub title="Wizards.cartovl.title" text="Wizards.cartovl.subtitle" tags="Wizards.cartovl.tags" iconmodifier="map"></selector-stub>
            <warning-stub></warning-stub>
          </div>
          <div class="grid-cell grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile">
            <selector-stub title="Wizards.cartoframes.title" text="Wizards.cartoframes.subtitle" tags="Wizards.cartoframes.tags" iconmodifier="notebook"></selector-stub>
            <warning-stub></warning-stub>
          </div>
          <div class="grid-cell grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile">
            <selector-stub title="Wizards.builder.title" text="Wizards.builder.subtitle" tags="Wizards.builder.tags" iconmodifier="window"></selector-stub>
          </div>
        </div>
        <p class="grid-cell u-mt--48 is-small">Wizards.Distributor.feedback.text
          <a href="https://docs.google.com/forms/d/e/1FAIpQLScBQUWd-TP3Qy514DOCNg-KoLrViHijUR5giLAMS-3jmDnrPg/viewform" target="_blank" rel="noopener noreferrer">
            Wizards.Distributor.feedback.link</a></p>
      </div>
    </div>
  </modal-stub>
  <router-view name="onboarding"></router-view>
</section>
`;

exports[`Distributor.vue should render properly 1`] = `
<section class="onboarding-welcome">
  <modal-stub name="onboarding-welcome" isopen="true">
    <div class="distributor gtm-onboarding">
      <div class="header">
        <div class="container">
          Wizards.Distributor.headerTitle
        </div>
      </div>
      <div class="container u-pt--36">
        <h1 class="u-pr--10 u-pl--10 u-mb--40 title is-body is-semibold">
          Wizards.Distributor.title
        </h1>
        <div class="grid u-flex__justify--between">
          <div class="grid-cell grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile">
            <selector-stub title="Wizards.cartovl.title" text="Wizards.cartovl.subtitle" tags="Wizards.cartovl.tags" iconmodifier="map"></selector-stub>
            <!---->
          </div>
          <div class="grid-cell grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile">
            <selector-stub title="Wizards.cartoframes.title" text="Wizards.cartoframes.subtitle" tags="Wizards.cartoframes.tags" iconmodifier="notebook"></selector-stub>
            <!---->
          </div>
          <div class="grid-cell grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile">
            <selector-stub title="Wizards.builder.title" text="Wizards.builder.subtitle" tags="Wizards.builder.tags" iconmodifier="window"></selector-stub>
          </div>
        </div>
        <p class="grid-cell u-mt--48 is-small">Wizards.Distributor.feedback.text
          <a href="https://docs.google.com/forms/d/e/1FAIpQLScBQUWd-TP3Qy514DOCNg-KoLrViHijUR5giLAMS-3jmDnrPg/viewform" target="_blank" rel="noopener noreferrer">
            Wizards.Distributor.feedback.link</a></p>
      </div>
    </div>
  </modal-stub>
  <router-view name="onboarding"></router-view>
</section>
`;
